src/js/components/List/List.js

```diff
@@ -357,13 +371,17 @@ const List = React.forwardRef(
                       // accessibility guidelines that focus remains on `ul`
                       listRef.current.focus();
                     },
-                    onMouseOver: () => setActive(index),
+                    onMouseOver: () => {
+                      setLastActive(undefined);
+                      setActive(index);
+                    },
                     onMouseOut: () => setActive(undefined),
                     onFocus: () => {
                       setActive(index);
                       setItemFocus(true);
                     },
                     onBlur: () => {
+                      setLastActive(active);
```

Comment:
I don't think we need to `setLastActive` here.

Reply:
After removal, functionality works as intended